### PR TITLE
Add ticket mutation tests

### LINKED ISSUE #2124: Add Tests for Domain Ticket Mutations
for the following files in `assets\prototype\domain\eventEditor\data\mutations\tickets`:

- updatePriceCache
- useOnCreateTicket
- useOnDeleteTicket
- useOnUpdateTicket
- useTicketMutator
- useUpdateTicketCache

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR:
- Adds Tests for domain ticket mutations (`create`, `update` and `delete`)
- Adds TS generics to some Ramda function calls
- Closes #2124

### CODE REVIEW

File: assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+const prices = { ...priceEdge, __typename: 'EspressoTicketPricesConnectionEdge' };
+
+export const mockedTickets = {
+	[MutationType.Create]: { ...tickets[0], id: tickets[0].id + '-alpha' }, // make sure to change the ID to make it different}
+	[MutationType.Update]: { ...tickets[0], prices },
+	[MutationType.Delete]: { id: tickets[0].id, __typename: tickets[0].__typename },
+};
+
+export const getMutationMocks = (mutationInput: MutationInput, mutationType: MutationType) => {
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts]: maybe type return value too ?

File: assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+	return {
+		query: mutations[`${mutationType}_TICKET`],
+		variables: {
+			input,
+		},
+	};
+};
+
+export const getMockResult = (mutationInput: MutationInput, mutationType: MutationType) => {
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts]: maybe type return value too ?

File: assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts (hunk 1 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/mutations/tickets/test/data.ts]: We need to have Apollo Query data defined for that. I will do that in a PR when I work on #2185

File: assets/prototype/domain/eventEditor/data/mutations/tickets/test/createTicket.test.ts [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+		// check if all the passed prices are related to the ticket
+		priceIds.forEach((ticketId) => {
+			const relatedTicketIds = mutationResult.current.relationsManager.getRelations({
+				entity: 'prices',
+				entityId: ticketId,
+				relation: 'tickets',
+			});
+
+			expect(relatedTicketIds).toContain(mockedTicket.id);
+		});
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/mutations/tickets/test/createTicket.test.ts]: :+1:

File: assets/prototype/domain/eventEditor/data/mutations/tickets/updatePriceCache.ts [hunk 4]
```diff
@@ -19,24 +19,30 @@ const updatePriceCache = ({ proxy, prices = {}, ticketIn, ticketId, remove = fal
 	try {
 		data = proxy.readQuery(queryOptions);
 	} catch (error) {
-		data = {};
+		data = {
+			espressoPrices: DEFAULT_ENTITY_LIST_DATA('Prices'),
```
Comment (nicolad) [inline assets/prototype/domain/eventEditor/data/mutations/tickets/updatePriceCache.ts]: I was expecting `DEFAULT_ENTITY_LIST_DATA` to be a constant/ object/ enum, not a function :)

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/data/mutations/tickets/updatePriceCache.ts]: Initially I defined it as a constant but then thought it's too much to define that for each entity.

**Pull request merged**